src/app/application.cpp

```diff
@@ -771,6 +797,51 @@ void Application::shutdownCleanup(QSessionManager &manager)
 }
 #endif
 
+#ifdef Q_OS_WIN
+void Application::applyMemoryWorkingSetLimit()
+{
+//    MEMORYSTATUSEX statex;
+//    statex.dwLength = sizeof(statex);
+//    if (!::GlobalMemoryStatusEx(&statex))
+//    {
+//        const DWORD errorCode = ::GetLastError();
+//        QString reason = tr("Error %1.").arg(errorCode);
+//        LPVOID lpMsgBuf = nullptr;
+//        if (::FormatMessageW((FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS)
+//                         , nullptr, ::GetLastError(), MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), reinterpret_cast<LPWSTR>(&lpMsgBuf), 0, nullptr))
+//        {
+//            reason = QString::fromWCharArray(reinterpret_cast<LPWSTR>(lpMsgBuf)).trimmed();
+//            ::LocalFree(lpMsgBuf);
+//        }
+//        LogMsg(tr("Failed to set Memory Working Set limit. Reason: %1").arg(reason), Log::WARNING);
+//        return;
+//    }
+
+    const int DEFAULT_WORKING_SET = 256;
+    const int MB = 1024 * 1024;
+    const std::int64_t limit = memoryWorkingSetLimit();
+    const SIZE_T maxSize = ((limit >= 0) ? limit : DEFAULT_WORKING_SET) * MB;
+    const SIZE_T minSize = (maxSize < (64 * MB)) ? maxSize : (64 * MB);
+    if (!::SetProcessWorkingSetSizeEx(::GetCurrentProcess(), minSize, maxSize, QUOTA_LIMITS_HARDWS_MAX_ENABLE))
+    {
+        const DWORD errorCode = ::GetLastError();
+        QString reason = tr("Error code: %1.").arg(errorCode);
+        LPVOID lpMsgBuf = nullptr;
+        if (::FormatMessageW((FORMAT_MESSAGE_ALLOCATE_BUFFER | FORMAT_MESSAGE_FROM_SYSTEM | FORMAT_MESSAGE_IGNORE_INSERTS)
+                         , nullptr, errorCode, MAKELANGID(LANG_NEUTRAL, SUBLANG_DEFAULT), reinterpret_cast<LPWSTR>(&lpMsgBuf), 0, nullptr))
+        {
+            QString message = QString::fromWCharArray(reinterpret_cast<LPWSTR>(lpMsgBuf)).trimmed();
+            ::LocalFree(lpMsgBuf);
+
+            if (!message.endsWith(u'.'))
+                message.append(u'.');
+            reason = u"Reason: %1 "_qs.arg(message) + reason;
```

Comment:
I would really add quotes around the error message from system, to avoid it blending in our log message.